RocketControlUnitGUI/src/data.ts

```diff
@@ -0,0 +1,110 @@
+  // data.ts
+  import PocketBase from 'pocketbase';
+  import dotenv from "dotenv";
+
+  
+
+  const ADMIN_EMAIL: string = import.meta.env.VITE_EMAIL;
+  const ADMIN_PASSWORD: string = import.meta.env.VITE_PASSWORD;
+
+ console.log(ADMIN_EMAIL);
+
+  const pb = new PocketBase('http://127.0.0.1:8090');
+  pb.autoCancellation(false);
+
+  let isAuthenticated = false;
+
+
+
+  // Function to authenticate the admin user
+  export async function authenticate() {
+    if (!isAuthenticated) {
+      await pb.admins.authWithPassword(ADMIN_EMAIL, ADMIN_PASSWORD);
+      isAuthenticated = true;
+    }
+  }
+
+  type RecordData = { [key: string]: any };
+  export type AllData = { [collectionName: string]: RecordData[] };
+
+  // Fetch paginated data with existing logic
+  export async function fetchPaginatedData(
```

Comment:
This should be extracted into its own hook that depends on `Return<typeof usePocketbase>`. Perhaps make a chart hook that encapsulates chart related logic to avoid global state for the chart system (this is not a big deal though, it would just be cleaner).